### ISSUE TITLE
Contact featurizer memoryError unspecified

### ISSUE THREAD

**Issue by orthonalmatrix** [msmbuilder/msmbuilder#1108]
Hi 
I've used the dihedralfeaturizer before with the same data but wanted to try the contactfeaturizer but if fails with an error that is unclear so I don't know how to proceed. The code is:
`
import mdtraj as md

from msmbuilder.io import gather_metadata, load_trajs, NumberedRunsParser, preload_tops, backup
from msmbuilder.featurizer import ContactFeaturizer, DihedralFeaturizer
from msmbuilder.feature_selection import VarianceThreshold
from msmbuilder.preprocessing import RobustScaler
from msmbuilder.decomposition import tICA
from msmbuilder.tpt import paths, net_fluxes
from msmbuilder.cluster import MiniBatchKMeans as cluster
from msmbuilder.msm import MarkovStateModel

import msmexplorer as msme
from matplotlib import pyplot as plt
%matplotlib inline

import numpy as np
import math

import seaborn as sns

import os
import sys

# Change/add numbers in this list to see the effects of other datasets on the msm model
used_trajs = [0,1,2]

#an initial load to get the indicies wanted
temp_traj = md.load("top.pdb")
#indicies = temp_traj.topology.select_expression("residue 26 to 493 and backbone") #just used to get the line below
#print(indicies)
#indicies = [atom.index for atom in temp_traj.topology.atoms if ((26 <= atom.residue.resSeq <= 493) and atom.is_backbone)]
indicies = [atom.index for atom in temp_traj.topology.atoms if ((26 <= atom.residue.resSeq <= 493))]
#print(indicies)

# Parses trajectory files for metadata
parser = NumberedRunsParser(
    #traj_fmt="s0tep7_productionC{run}.nowat.dcd",
    traj_fmt="traj{run}.dcd",
    top_fn="trajs/nowat.pdb",
    step_ps=20,
)

# Gathers metadata using parser
#meta = gather_metadata("trajs/s0tep7*nowat.dcd", parser)
meta = gather_metadata("trajs/traj*.dcd", parser)

# Loads topology (.pdb) file
tops = preload_tops(meta)

# Load all trajectories
trajs = []
for i, row in meta.iterrows():
    trajs.append(md.load(row['traj_fn'], top=tops[row['top_fn']],atom_indices=indicies))
    #trajs.append(md.load(row['traj_fn'], top=tops[row['top_fn']]))

#load the full trajectory for output
fulltrajs = []
for i, row in meta.iterrows():
    fulltrajs.append(md.load(row['traj_fn'], top=tops[row['top_fn']] ))

trajs_slice = trajs

feat_sel = 'Contact'

# Featurize selected, sliced trajectory
feat_dict = {'Contact': ContactFeaturizer(),'Dihedral': DihedralFeaturizer()}

feat = feat_dict[feat_sel]

#feats_orig = feat.transform(trajs_slice)
feats_orig = feat.transform(trajs)

# Define a subset of the transforamed trajectory to analyze
# The larger set will be used for testing
feats = [frames[::10] for frames in feats_orig]
`

The error is:
`---------------------------------------------------------------------------
MemoryError                               Traceback (most recent call last)
<ipython-input-7-40b6950a9007> in <module>
      7 
      8 #feats_orig = feat.transform(trajs_slice)
----> 9 feats_orig = feat.transform(trajs)
     10 
     11 # Define a subset of the transforamed trajectory to analyze

~/bin/anaconda3/envs/msm/lib/python3.6/site-packages/msmbuilder/featurizer/featurizer.py in transform(self, traj_list, y)
    158             (n_samples_i, n_features)
    159         """
--> 160         return [self.partial_transform(traj) for traj in traj_list]
    161 
    162     def describe_features(self, traj):

~/bin/anaconda3/envs/msm/lib/python3.6/site-packages/msmbuilder/featurizer/featurizer.py in <listcomp>(.0)
    158             (n_samples_i, n_features)
    159         """
--> 160         return [self.partial_transform(traj) for traj in traj_list]
    161 
    162     def describe_features(self, traj):

~/bin/anaconda3/envs/msm/lib/python3.6/site-packages/msmbuilder/featurizer/featurizer.py in partial_transform(self, traj)
   1085         else:
   1086             distances, _ = md.compute_contacts(traj, self.contacts,
-> 1087                                                self.scheme, self.ignore_nonprotein)
   1088         return self._transform(distances)
   1089 

~/bin/anaconda3/envs/msm/lib/python3.6/site-packages/mdtraj/geometry/contact.py in compute_contacts(traj, contacts, scheme, ignore_nonprotein, periodic, soft_min, soft_min_beta)
    206             n_atom_pairs_per_residue_pair.append(residue_lens[pair[0]] * residue_lens[pair[1]])
    207 
--> 208         atom_distances = md.compute_distances(traj, atom_pairs, periodic=periodic)
    209 
    210         # now squash the results based on residue membership

~/bin/anaconda3/envs/msm/lib/python3.6/site-packages/mdtraj/geometry/distance.py in compute_distances(traj, atom_pairs, periodic, opt)
     71         orthogonal = np.allclose(traj.unitcell_angles, 90)
     72         if opt:
---> 73             out = np.empty((xyz.shape[0], pairs.shape[0]), dtype=np.float32)
     74             _geometry._dist_mic(xyz, pairs, box.transpose(0, 2, 1).copy(), out, orthogonal)
     75             return out

MemoryError: 
`

It seems odd that there is nothing after the memortError.
I am running in jupyter-lab but nothing shows up in the terminal either as an error. I appreciate any help!

**Comment by cxhernandez** [comment]
I believe you've run out of memory on your machine. How large is your dataset relative to the amount of memory allocated? Does this run with a smaller dataset?

**Comment by orthonalmatrix** [comment]
Hi. Thanks for replying. I don't think that is the case. I have 40 Gb on the machine and have no trouble with the dihedralFeaturizer (although I guess the contact could require more memory). But my system memory allocation does not increase as it is running  and the file is small:
-rw-rw-r-- 1 eddie eddie 275M Aug 27 15:31 ../p450_apo_C7/step7_production_apo_C7.nowater.dcd
(this is the reduced data set for testing). Given all this, do you think it could be memory?

**Comment by cxhernandez** [comment]
I have no reason to not suspect it. You can use `%memit` or a command-line tool like `htop` to profile your memory usage during featurization. If it doesn't seem like available memory is an issue, then I'd raise an issue on the MDTraj repo as a potential memory leak.

**Comment by eackad** [comment]
I have looked at the memory through the system resource monitorin ubuntu. Looked like there was plenty of room. What is very odd is I could use the dihedral featurizure or the DRID with the full data set (~4Gb) and they worked fine. Is the contact featurizer more memory intensive?

**Comment by cxhernandez** [comment]
Yes, but some more than others (e.g. `closest`). One last thing to try is a newer version of MDTraj. For unit testing purposes, we've pinned it to v1.8 but you can try upgrading to v1.9 and see if the issue persists. If it does, again, I recommend posting an issue to the mdtraj repo, where you can get better support for this particular issue.

**Comment by orthonalmatrix** [comment]
I am using mdtraj  1.9.2. I tried 1.9.1 and still had the same issue. I'll ask the mdtraj people.
